### PR TITLE
Update Transformer.write* to use the new object-style api

### PR DESCRIPTION
**Pull request by dannyvv**

### CODE REVIEW

**Comment by dannyvv** [comment]
This PR will have to wait untill we have updated the Public LKG and deployed it to cloudbuild.

File: Examples/DotNetCoreBuild/test/TestDirectories/testDirectories.dsc [hunk 1]
```diff
@@ -56,7 +56,10 @@ export const writeHardLinkInSharedOpaqueDirectoryIsAllowed = !Context.isWindowsO
     linkFileIntoDirectory(f`testDirectories.dsc`, sod);
 
     // hardlink an output file
-    const outFile = Transformer.writeFile(p`${Context.getNewOutputDirectory("write-file")}/writefile.txt`, "hi");
+    const outFile = Transformer.writeAlLText({
```
Comment (narasamdya) [inline Examples/DotNetCoreBuild/test/TestDirectories/testDirectories.dsc]: >writeAlLText [](start = 32, length = 12)

typo?

**Comment by narasamdya** [comment]
Is it new feature? Or is it replacing the old one? Are you going to take care of necessary changes in Office SDK?

**Comment by dannyvv** [comment]
> Is it new feature? Or is it replacing the old one? Are you going to take care of necessary changes in Office SDK?

When rolled out everywhere I'll follow up with all customers and see if I can update them. When done I hope to retire the old api if possible, or at least mark Obsolete

File: Examples/DotNetCoreBuild/test/TestDirectories/testDirectories.dsc (hunk 1 above)
Comment (dannyvv) [reply Examples/DotNetCoreBuild/test/TestDirectories/testDirectories.dsc]: > > writeAlLText [](start = 32, length = 12)
> 
> typo?
> If yes, how could this test have passed?

It was a typo and it didn't pass /partB validation.

**Pull request merged**